sklearn/tests/test_pipeline.py

```diff
@@ -227,7 +228,8 @@ def test_pipeline_invalid_parameters():
         pipe.set_params(anova__C=0.1)
 
     # Test clone
-    with pytest.warns(None):
+    with warnings.catch_warnings():
+        warnings.simplefilter("error", UserWarning)
         pipe2 = clone(pipe)
```

Comment:
From the PR that added this check: https://github.com/scikit-learn/scikit-learn/pull/9716, I think this check was to catch a deprecation warning in 0.18 that is no longer in the library.

With that in mind, I think we can remove the check all together:

```suggestion
    pipe2 = clone(pipe)
```

Reply:
Thanks for the suggestion @thomasjpfan. I will adopt your suggestion. Hope I could get more familiar with the codebase after making some contributions.